packages/analytics-js/src/components/errorListener/windowOnError.ts

```diff
@@ -0,0 +1,88 @@
+import type { IPluginsManager } from '@rudderstack/analytics-js-common/types/PluginsManager';
+import type { ILogger } from '@rudderstack/analytics-js-common/types/Logger';
+import { state } from '../../state';
+import type { ErrorMetaData, JqueryErrorEvent } from './types';
+
+const attachOnError = (pluginsManager: IPluginsManager, logger?: ILogger) => {
+  const prevOnError = (globalThis as any).onerror;
+
+  function onerror(
```

Comment:
Is a majority of this code copied from somewhere? Let's not do that. It could cause licensing issues.
It's better to rewrite the logic. Moreover, many of the error scenarios are probably not applicable to us.

Reply:
Also, let's not use code that we don't fully understand.

Reply:
https://github.com/bugsnag/bugsnag-js/blob/next/LICENSE.txt no licencing issues.

What are the parts we do not nderstand Do we need a meeting here?

Reply:
The license says we need to include their copyright notice. I'm not certain but should we include it for this kind of snippets we take from their software?

Reply:
I believe some corner cases and the JQuery error handling are irrelevant.
IMO, the logic in these handlers should be straightforward as we later filter errors that do not originate from the SDK.